crates/nu-protocol/src/config/mod.rs

```diff
@@ -219,374 +162,333 @@ impl Value {
+                                    _ => {
+                                        report_invalid_value("expected a unsigned integer or a record", span, &mut errors);
+                                        *value = config.table.padding.into_value(span);
                                     }
-                                    "abbreviated_row_count" => {
-                                        if let Ok(b) = value.as_int() {
-                                            if b < 0 {
-                                                report_invalid_value("should be an int unsigned", span, &mut errors);
-                                            }
-
-                                            config.table_abbreviation_threshold = Some(b as usize);
-                                        } else {
-                                            report_invalid_value("should be an int", span, &mut errors);
+                                },
+                                "index_mode" => {
+                                    process_string_enum(
+                                        &mut config.table.index_mode,
+                                        &[key, key2],
+                                        value,
+                                        &mut errors
+                                    );
+                                }
+                                "trim" => {
+                                    match try_parse_trim_strategy(value, &mut errors) {
+                                        Ok(v) => config.table.trim = v,
+                                        Err(e) => {
+                                            // try_parse_trim_strategy() already adds its own errors
+                                            errors.push(e);
+                                            *value = config.table.trim.clone().into_value(span);
                                         }
                                     }
-                                    _ => {
-                                        report_invalid_key(&[key, key2], span, &mut errors);
-                                        return false
+                                }
+                                "show_empty" => {
+                                    process_bool_config(value, &mut errors, &mut config.table.show_empty);
+                                }
+                                "abbreviated_row_count" => {
+                                    if let Ok(b) = value.as_int() {
+                                        if b < 0 {
+                                            report_invalid_value("should be an int unsigned", span, &mut errors);
+                                        }
+
+                                        config.table.abbreviated_row_count = Some(b as usize);
+                                    } else {
+                                        report_invalid_value("should be an int", span, &mut errors);
```

Comment:
This is missing the necessary error recovery and nevertheless writes the truncating cast after pushing the error (forgot that in #10896)

With the few places it appears in, maybe we reach the threshold for an unsigned int helper?

Reply:
Fixed!

> With the few places it appears in, maybe we reach the threshold for an unsigned int helper?

Already done in a different branch, for a future PR lol.

Reply:
To clean up the value can you set it to what is in `Config` after `report_invalid_value`

Reply:
Fixed, if that looks right.

Reply:
Based on this logic we also need to accept `null` as a value which sets the internal back to `None`. In the default config this entry is commented out. so never set like that.